### PR TITLE
Remap meta mappings for extensive `type` usage

### PR DESCRIPTION
**Pull request by sc0ttkclark**
Added new dynamic template mapping for long, double, date, datetime, time as well -- same as #372

- [x] Unit test for all `'type'` options for `'meta_query'`
- [x] README.md documentation update
- [x] Update `@since` if needed

### CODE REVIEW

**Comment by tlovett1** [comment]
Is this ready to go?

**Comment by sc0ttkclark** [comment]
Needs tests for each 'meta_query' 'type' option that WP_Query currently supports, to ensure they work as expected for ES (no unexpected exceptions, etc).

File: classes/class-ep-api.php [hunk 1]
```diff
@@ -489,6 +489,9 @@ public function prepare_post( $post_id ) {
 		 * This filter is named poorly but has to stay to keep backwards compat
 		 */
 		$post_args = apply_filters( 'ep_post_sync_args', $post_args, $post_id );
```
Comment (tlovett1) [inline classes/class-ep-api.php]: Let's create another similarly named filter after `prepare_meta_types`

**Comment by tlovett1** [comment]
This looks really solid. Had one comment.

**Comment by tlovett1** [comment]
We will need lots of documentation updates for this.

**Comment by ChrisWiegman** [comment]
I've added unit tests to test the prep function. I'm thinking of adding an "object" type here to cover data structures. What do you think @sc0ttkclark ?

**Comment by tlovett1** [comment]
@ChrisWiegman if object actually works I think that would be cool. The sense that I got was that it isn't possible to index arbitrary objects.

**Comment by sc0ttkclark** [comment]
From what I gathered, I didn't think objects could be used here because you had to define the specific fields on those objects.

**Comment by sc0ttkclark** [comment]
I also don't think serialized data should be evaluated in this case, unless it was a basic object already (serialized arrays don't sound super easy to interact with in ES)

**Pull request merged**